trade-finance-manager-api/src/v1/controllers/deal.tasks.js

```diff
@@ -2,22 +2,6 @@ const api = require('../api');
 const CONSTANTS = require('../../constants');
 const { createTasks } = require('../helpers/create-tasks');
 
-/**
- * Check if the "create or match parties" task should be created
- * @param {Object} deal
- * @returns {Boolean}
- */
-const shouldCreatePartiesTask = (deal) => {
-  const { tfm } = deal;
```

Comment:
I'm afraid removing a whole function alongside with the test is not ideal.
What is requires is as follows:

1. Re-instate the function.
2. Introduce conditional branching for `BSS/EWCS` and `GEF` deals.
3. If `BSS/EWCS` ensure both `exporter` and `buyer` Party URN have been inputed by the TFM, other only `exporter` if GEF.
4. Once the conditions have been met then return `false` otherwise `true`.
5. Re-introduce the API level tests with extended coverage please.

Thanks

Reply:
done